client/galaxy/scripts/components/Workflow/Editor/Recommendations.vue

```diff
@@ -25,6 +25,7 @@
 import { getToolPredictions } from "./services";
 import LoadingSpan from "components/LoadingSpan";
 import _l from "utils/localization";
+import _ from "underscore";
```

Comment:
@anuprulez is there no way to not use underscore? We want to get rid of this library in the long run and just use standard JS.

Also can you make sure you upstream the changes? Thanks.

Reply:
Ok, the changes will be done in the upstream. Do you know why "underscoreJS" will be removed? @bgruening

Reply:
I think the idea is to totally get rid of underscore and jQuery if possible as most of the functionality is now available in ES6.

Reply:
Yes, it makes sense. Thanks!

Reply:
@anuprulez btw. why do you filter on the client? Should the API not filter that and __not__ return the duplicates?

Reply:
The API returns a unique list but the datatype compatibility is checked on the client-side (I use the same logic which is being used to check datatype compatibility while connecting toolboxes on the workflow editor i.e. showing orange or green colour). This part of the code needs to be refactored which erroneously adds duplicate items while checking for compatibility between all input (of all the recommended tools) and output datatypes (of the last tool).

Reply:
And this part of the code is currently separate for tool recommendation shown after a tool execution (here this case is handled) and in the workflow editor (here it was not). This needs to be unified. I will do all this refactoring and push it to upstream.